src/main/scala/cromwell/engine/ExecutionStore.scala

```diff
@@ -0,0 +1,37 @@
+package cromwell.engine
+
+import cromwell.binding.{Call, WdlBinding}
+import cromwell.engine.ExecutionStatus.ExecutionStatus
+import scala.collection.mutable
+
+object ExecutionStatus extends Enumeration {
+  type ExecutionStatus = Value
+  val NotStarted, Running, Failed, Done = Value
+}
+
+/**
+ * Corresponds to the "execution table" of our discussions.
+ */
+class ExecutionStore(binding: WdlBinding) {
+
+  def isDone: Boolean = !table.exists(_._2 != ExecutionStatus.Done)
```

Comment:
If you're of the camp that tries to avoid nots in booleans, this is equivalent:

    table.find(_._2 != ExecutionStatus.Done).isEmpty